__tests__/app.test.js

```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
+  test("400: missing a 'body' key from the client's request", () => {
+    const article_id = 2;
+    const newComment = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: missing a username from the client's request (value)", () => {
+    const article_id = 2;
+    const newComment = {
+      username: "",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: PSQL would not allow to post comments on a non-existent article, as it violates foreign key constraint", () => {
+    const article_id = 999;
+    const newComment = {
+      username: "butter_bridge",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: invalid data type requested by the client (string)", () => {
+    const article_id = "banana";
+    const newComment = {
+      username: "butter_bridge",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: invalid data type requested by the client (negative integer)", () => {
+    const article_id = -12;
+    const newComment = {
+      username: "butter_bridge",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: invalid data type requested by the client (float)", () => {
+    const article_id = 3.5;
+    const newComment = {
+      username: "butter_bridge",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: username requested by the client is not in the database", () => {
+    const article_id = 2;
+    const newComment = {
+      username: "peter_griffin",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
```

Comment:
Please write another test (404) where the article ID is valid data type, but there's no article with that ID.